packages/actor-rdf-resolve-hypermedia-links-queue-wrapper-info-occupancy/lib/LinkQueueLogger.ts

```diff
@@ -0,0 +1,164 @@
+   */
+  private static getActorProductorInformation(link: ILink): IProducedByActor | null {
+    const metadata = link.metadata;
+    if (metadata !== undefined && metadata[PRODUCED_BY_ACTOR] !== undefined) {
+      const { name, ...rest } = metadata[PRODUCED_BY_ACTOR];
+
+      if (name === undefined) {
+        return null;
+      }
+      return {
+        name,
+        metadata: Object.keys(rest).length === 0 ? undefined : rest,
+      };
+    }
+    return null;
+  }
+
+  /**
+   * Update the link production ratio
+   * @param {IUrlStatistic} link - current link
+   * @param {keyof ILinkProductionActorRatio} event - link queue event
+   */
+  private updateLinkProductionRatio(link: IUrlStatistic, event: keyof ILinkProductionActorRatio): void {
+    if (link.producedByActor &&
+      this.linkProductionRatio[event][link.producedByActor?.name]) {
+      this.linkProductionRatio[event][link.producedByActor?.name] += 1;
+    } else if (
+      this.linkProductionRatio[event][link.producedByActor?.name ?? 'unknown'] === undefined) {
+      this.linkProductionRatio[event][link.producedByActor?.name ?? 'unknown'] = 1;
+    } else {
+      this.linkProductionRatio[event].unknown += 1;
+    }
+  }
+
+  /**
+   * Create the information of a link queue event
+   * @param {ILink} link - current link
+   * @param {EventType} eventType - the type of event
+   * @param {ILink|undefined} parent - the parent of the link
+   * @returns {ILinkQueueEvent} current event of the link queue
+   */
+  private createLinkQueueEvent(link: ILink, eventType: 'PUSH' | 'POP', parent?: ILink): ILinkQueueEvent {
+    const linkInfo: IUrlStatistic = {
+      url: link.url,
+      producedByActor: LinkQueueLogger.getActorProductorInformation(link),
+      timestamp: performance.now(),
+      parent: parent?.url,
+    };
+    this.updateLinkProductionRatio(linkInfo, eventType === 'POP' ? 'popEvents' : 'pushEvents');
+
+    return {
+      type: eventType,
+      link: linkInfo,
+      query: this.query,
+      queue: {
+        size: this.getSize(),
+        ...this.linkProductionRatio,
+      },
+    };
+  }
+
+  public override push(link: ILink, parent: ILink): boolean {
+    const resp: boolean = super.push(link, parent);
+    if (resp) {
+      const event: ILinkQueueEvent = this.createLinkQueueEvent(link, 'PUSH', parent);
+      this.materialize(event);
+    }
+    return resp;
+  }
+
+  public override pop(): ILink | undefined {
+    const link = super.pop();
+    if (link !== undefined) {
+      const event: ILinkQueueEvent = this.createLinkQueueEvent(link, 'POP');
+      this.materialize(event);
+    }
+    return link;
+  }
+
+  /**
+   * Materialize the current event into the logger
+   * @param {ILinkQueueEvent} event - Current event
+   */
+  private materialize(event: ILinkQueueEvent): void {
+    const jsonEvent = { ...event, type: event.type };
+    this.logger.trace(LinkQueueLogger.LINK_QUEUE_EVENT_NAME, { data: JSON.stringify(jsonEvent) });
```

Comment:
Is the `stringify` needed? I think the logger already does stringification internally.

Reply:
I will check, but I think I did it because the engine was crashing if I was not doing it, maybe it was related to the metadata in the query that couldn't be transformed into a string.

Reply:
I remember why it is because it is difficult for another program to read interpret the output if it is a JSON object and not a JSON string.

Reply:
Sounds like a problem in your other program then 😉

Reply:
But, I don't know how to parse a JSON object in a string format that is not "stringify". JSON.parse doesn't seems to work.